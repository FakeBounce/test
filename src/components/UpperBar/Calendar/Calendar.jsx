import React, { Component } from "react";
import styles from "./Calendar.module.scss";
import moment from "moment";

class Calendar extends Component {
  lowerCaseAllWordsExceptFirstLetters = string => {
    return string
      .toLowerCase()
      .split(" ")
      .map(s => s.charAt(0).toUpperCase() + s.substring(1))
      .join(" ");
  };

  render() {
    return (
      <div className={styles.CalendarContainer}>
        <div>
          <p>{moment().format("dddd")}</p>
          <p>
            {this.lowerCaseAllWordsExceptFirstLetters(
              moment().format("D MMMM YY")
            )}
          </p>
        </div>
      </div>
    );
  }
}
export default Calendar;
