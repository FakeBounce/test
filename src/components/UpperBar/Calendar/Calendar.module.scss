@import '~common/styles/variables';
.CalendarContainer {
  border-right: 1px solid rgba(151, 151, 151, 0.54);
  order: 1;
  display: flex;
  align-items: center;
}

.CalendarContainer > div {
  display: flex;
  text-align: right;
  flex-direction: column;
  padding-right: 15px;
}

.CalendarContainer > div > p {
  // font-size: 12px;
  // color: $marine-blue;
  margin: 0;
  color: #9FA9B5;
  font-size: 13px;
  font-weight: 300;
  line-height: 16px;
  text-align: right;
  letter-spacing: normal;
  &:first-child {
    text-transform: capitalize;
  }

  
}