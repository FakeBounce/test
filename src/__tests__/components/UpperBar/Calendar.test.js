import React from "react";
import { shallow, configure } from "enzyme/build";
import Adapter from "enzyme-adapter-react-16/build";
import Calendar from "components/UpperBar/Calendar/Calendar";

configure({ adapter: new Adapter() });

// Return a fixed timestamp when moment().format() is called
jest.mock("moment", () => () => ({ format: () => "2018–01–30" }));

describe("Testing Calendar component", () => {
  it("renders as expected", () => {
    const wrapper = shallow(<Calendar />);
    expect(wrapper).toMatchSnapshot();
  });
});
