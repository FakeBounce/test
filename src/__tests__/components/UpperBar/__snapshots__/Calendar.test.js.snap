// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Calendar component renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Calendar />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div>
        <p>
                2018–01–30
        </p>
        <p>
                2018–01–30
        </p>
</div>,
      "className": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            2018–01–30
</p>,
          <p>
            2018–01–30
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018–01–30",
          },
          "ref": null,
          "rendered": "2018–01–30",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "2018–01–30",
          },
          "ref": null,
          "rendered": "2018–01–30",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div>
          <p>
                    2018–01–30
          </p>
          <p>
                    2018–01–30
          </p>
</div>,
        "className": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p>
              2018–01–30
</p>,
            <p>
              2018–01–30
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018–01–30",
            },
            "ref": null,
            "rendered": "2018–01–30",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "2018–01–30",
            },
            "ref": null,
            "rendered": "2018–01–30",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Array [],
  Symbol(enzyme.__childContext__): null,
}
`;
